src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -1021,8 +1025,49 @@ public RequestLineStatus TakeStartLine(SocketInput input)
                     scan.Skip(method.Length);
                 }
 
-                scan.Take();
+                scan.Take(); // consume space
+
+                // begin consuming request-target
                 begin = scan;
+
+                var targetIsAbsolute = false;
+                string requestUriScheme;
```

Comment:
still not sure why you're creating these strings only to rebuild the rawTarget later. why not capture the full rawTarget all at once?